### ISSUE TITLE
Enhancement: hazelcast-jet-sql, time bounded query against Kafka

### ISSUE THREAD

**Issue by neilstevenson** [hazelcast/hazelcast-jet#2693]
When querying a streaming source such as a Kafka topic, it would be useful to have a time bound to terminate the query

Eg `SELECT * FROM public topicname LIMIT 5 seconds`

So the result set closes if no more messages arrive in 5 seconds. 
This would facilitate batch style processing of infinite sources.

Related, https://github.com/hazelcast/hazelcast-jet/issues/2692

**Comment by neilstevenson** [comment]
A time based cut-off would be useful for a query against batch too, if it runs too long

**Comment by viliam-durina** [comment]
We already have `SqlStatement.setTimeoutMillis`, but it's ignored for Jet. And even if it was not, the query will end with a `TimeoutException`, that's not ok IMO.

**Comment by neilstevenson** [comment]
Yes indeed, there's a difference here between batch and streaming. For batch the timebound would be to control erroneous queries, so to exceed the time limit would be an error. For streaming, it's more a point to consider the result set complete, we wish to wait for no more results.
